frontend/src/components/register/registerForm/RegisterForm.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * Form for email, password and repeated password to register new user
+ */
+
+import React from "react";
+
+import { createStyles, makeStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+
+import useObservable from "../../../hooks/observable";
+import { RegisterState } from "./utils/interfaces";
+import RegisterError from "./RegisterError";
+import RegisterSuccess from "./RegisterSuccess";
+import EmailField from "./RegisterFormEmail";
+import PasswordField from "./RegisterFormPassword";
+
+import authService from "../../../services/auth";
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    button: {
+      marginTop: "3%",
+      width: "85%"
+    }
+  })
+);
+
+/**
+ * Form with email, password and repeated password
+ */
+function RegisterForm() {
+  const classes = useStyles();
+
+  const [registerData, setRegisterData] = React.useState<RegisterState>({
+    email: "",
+    password: "",
+    password2: ""
+  });
+
+  const [error, setError] = React.useState(false);
```

Comment:
Errors should be assigned per field. Try do it this way:
``` 
  const registerInitialState = {
    email: "",
    password: "",
    password2: ""
  }
  const [registerData, setRegisterData] = React.useState<RegisterState>(registerInitialState);
  cosnt  [registerErrors, setRegisterErrors] = React.useState<RegisterState>(registerInitialState);

  const getError = {
     email: email => !!email  // validation function here. Return "" if everything is ok. Check if email is correct",
     password: password => !!password // Check if password is long enough"",
     password2: password2 => !!password2 // Check if password2 matches password""
  }

  const handleChange = (prop: keyof RegisterState) => (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setRegisterData({ ...registerData, [prop]: event.target.value });
    setRegisterErrors({ ...registerErrors, [prop]: getError[prop]( event.target.value ) });
  };



```

Reply:
This will also reset the errors when you fix the input

Reply:
Done. Something similar.